core/rawdb/accessors_chain.go

```diff
@@ -702,6 +703,95 @@ func DeleteBlockWithoutNumber(db ethdb.KeyValueWriter, hash common.Hash, number
 	DeleteTd(db, hash, number)
 }
 
+const badBlockToKeep = 10
+
+type badBlock struct {
+	Header *types.Header
+	Body   *types.Body
+}
+
+// badBlockList implements the sort interface to allow sorting a list of
+// bad blocks by their number in the reverse order.
+type badBlockList []*badBlock
+
+func (s badBlockList) Len() int { return len(s) }
+func (s badBlockList) Less(i, j int) bool {
+	return s[i].Header.Number.Uint64() > s[j].Header.Number.Uint64()
+}
```

Comment:
This is a bit un-intuitive. The `less(i,j)` - function should be the other way around, and you should use `sort.Reverse(...)` if you want it in reverse order, rather than changing how the `Less`-function is defined

Reply:
fixed